### DISCUSSION TITLE
Can I specify multiple repository locations at the same time?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [rustic-rs/rustic#1147] [Q&A] [answered]
I want to create the snapshots to several locations but I don't want to have multiple repositories. What I'm looking for is a way to specify 2 or more locations/backends, something like two rclone locations in the cloud, so every command will affect all locations as I can't really sync both as they are in the cloud, this way rustic will calculate the diff once and upload it to all the locations at the same time.

**Comment by aawsome** [comment, marked as answer]
This is currently not supported by rustic. But AFAIK rclone does support such a setting. That means you might configure rclone to use multiple backends and use the rclone backend in rustic. But I never tested or read docu about that.

What is supported: A possibility to use a hot repo part and a cold repo part. rustic copies hot data to both parts. This could be extended in future to multiple location, but there are many use cases which should then be also tackled like
- error handling: What if one location throws an error but other(s) could be available?
- checking: Do we want/need to check all locations when we run `check`

So, this is future music and currently very low on the improvement list of rustic.

**Reply by parasiteoflife** [reply]
Thank you!
I understand this isn't priority but would be great to have to replicate the repo to several local drives too. I will look into using rclone for this.

**Reply by kapitainsky** [reply]
Even if in theory you could use `rclone union` to implement sort of RAID1 using multiple remotes I would strongly advise not to do it. The same reason as @aawsome already mentioned in context of implementing it in `rustic` - errors handling and checking. Such solution would only work when  everything works 100% errors free. Proper design should never rely on such assumption. If I was you I would simply run multiple `rustic` backups  against the same local file system snapshot (it guarantees that the same data is used ) - you do this in parallel or in sequence depends on your network uplink vs remote speed capacity mostly.
